File: backend/eventhive/src/main/java/edu/cit/resuera/eventhive/event/dto/EventResponse.java
package edu.cit.resuera.eventhive.dto;

import java.time.LocalDateTime;

import edu.cit.resuera.eventhive.entity.EventStatus;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class EventResponse {
    private Long id;
    private String title;
    private String description;
    private LocalDateTime startDate;
    private LocalDateTime endDate;
    private String location;
    private String category;
    private String imageUrl;
    private Integer maxParticipants;
    private EventStatus status;
    private Long organizerId;
    private String organizerName;
    private LocalDateTime createdAt;
    private Integer participantCount;
    private Boolean isRegistered;
}

File: mobile/app/src/main/java/com/resuera/eventhive/features/events/EventResponse.kt
package com.resuera.eventhive.features.events

data class EventResponse(
    val id: Long,
    val title: String,
    val description: String?,
    val startDate: String?,
    val endDate: String?,
    val location: String?,
    val category: String?,
    val imageUrl: String?,
    val maxParticipants: Int?,
    val status: String?,
    val organizerId: Long?,
    val organizerName: String?,
    val createdAt: String?,
    val participantCount: Int?,
    val isRegistered: Boolean?
)